### PR TITLE
Update ElasticSearch indexing to 2.3.3.

### PR DESCRIPTION
**Pull request by ColinArenz**

### CODE REVIEW

File: indexing/src/main/scala/org/allenai/common/indexing/BuildCorpusIndex.scala [hunk 1]
```diff
@@ -61,10 +61,7 @@ class BuildCorpusIndex(config: Config) extends Logging {
       val createIndexRequestBuilder: CreateIndexRequestBuilder =
         esClient.admin().indices().prepareCreate(indexName)
 
-      // Optimize settings for indexing
```
Comment (ColinArenz) [inline indexing/src/main/scala/org/allenai/common/indexing/BuildCorpusIndex.scala]: We don't want to optimize for indexing in this way.  Shard count can't be changed after creating an index, and the new cluster is optimized for load balancing across a few nodes.  Current defaults are 5 shards, 1 replica, which provides pretty good load balancing and allows us to add/remove/replace nodes on a whim without snapshot/restore.

**Comment by ColinArenz** [comment]
@OyvindTafjord FYI

**Comment by ColinArenz** [comment]
Switching over to Oyvind for review due to vacation schedule.

**Comment by OyvindTafjord** [comment]
LGTM! Merging.

**Pull request merged**